### PR TITLE
UI: Keep mobile drawer open on component selection

### LINKED ISSUE #24255: On the sidebar, make it toggle when clicking on a component

### LINKED ISSUE #24256: Change the background of the sidebar on mobile

### PR DESCRIPTION
**Pull request by cdedreuille**
Closes https://github.com/storybookjs/storybook/issues/24255
Closes https://github.com/storybookjs/storybook/issues/24256

## What I did

On the new mobile navigation we debated on the best way to close the main navigation when selected a story. We decided that the drawer will only close if a story is selected, not when a component is toggled. The behaviour remain unchanged on desktop.

I also improved a few styles on the sidebar as well as the about section on mobile

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [x] end-to-end tests

#### Manual testing

- Run Storybook UI locally on your phone
- Open the main navigation
- Press on a component
- It should open the list of stories but not select any stories

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/ui/manager/src/components/upgrade/UpgradeBlock.tsx [hunk 1]
```diff
@@ -40,10 +41,13 @@ const Container = styled.div(({ theme }) => ({
   borderRadius: 5,
   padding: 20,
   marginTop: 0,
-  maxWidth: 400,
   borderColor: theme.appBorderColor,
   fontSize: theme.typography.size.s2,
   width: '100%',
+
+  [`@media ${BREAKPOINT_MIN_600}`]: {
```
Comment (JReinhold) [inline code/ui/manager/src/components/upgrade/UpgradeBlock.tsx:48]: Should we have a constant that is this whole string, like `IS_MOBILE_BREAKPOINT`?

**Review by JReinhold — CHANGES_REQUESTED** [review]
Code wise this looks good, I'll test the functionality tomorrow.

We want tests for this, either stories or e2e - we can work together on this. I see in the PR description you've checked off all the tests, but that's not quite right though, so maybe you've misunderstood that part?

I also think it's time to move `isDesktop` to the Mobile provider, so we don't keep creating change listeners all over the place. Feel free to do that here or in a follow up PR

File: code/ui/manager/src/components/upgrade/UpgradeBlock.tsx (hunk 1 above)
Comment (cdedreuille) [reply code/ui/manager/src/components/upgrade/UpgradeBlock.tsx]: I chose to do:
```tsx
export const BREAKPOINT = 600;
export const MEDIA_MIN_BREAKPOINT = `@media (min-width: ${BREAKPOINT}px)`;
```

And in the layout provider:
```tsx
const isDesktop = useMediaQuery(`(min-width: ${BREAKPOINT}px)`);
const isMobile = !isDesktop;
```

**Comment by cdedreuille** [comment]
@JReinhold I made the change on the layout provider to include `isDesktop` and `isMobile`. Let's work together tomorrow on these e2e tests. 🙏

File: code/ui/manager/src/components/upgrade/UpgradeBlock.tsx (hunk 1 above)
Comment (cdedreuille) [reply code/ui/manager/src/components/upgrade/UpgradeBlock.tsx]: Change to `MEDIA_DESKTOP_BREAKPOINT`

File: code/ui/manager/src/components/sidebar/Tree.tsx [hunk 2]
```diff
@@ -231,6 +234,7 @@ const Node = React.memo<NodeProps>(function Node({
           onClick={(event) => {
             event.preventDefault();
             onSelectStoryId(item.id);
+            if (!isDesktop) setMobileMenuOpen(false);
```
Comment (JReinhold) [inline code/ui/manager/src/components/sidebar/Tree.tsx:237]: Let's use `isMobile` instead here.

File: code/ui/manager/src/components/sidebar/Menu.tsx [hunk 3]
```diff
@@ -115,7 +114,7 @@ export interface SidebarMenuProps {
 
 export const SidebarMenu: FC<SidebarMenuProps> = ({ menu, isHighlighted, onClick }) => {
   const [isTooltipVisible, setIsTooltipVisible] = useState(false);
-  const isDesktop = useMediaQuery(BREAKPOINT_MIN_600);
+  const { isDesktop } = useLayout();
 
   if (!isDesktop) {
```
Comment (JReinhold) [inline code/ui/manager/src/components/sidebar/Menu.tsx:119]: `isMobile` here instead

Comment (cdedreuille) [reply code/ui/manager/src/components/sidebar/Menu.tsx]: Made the change

File: code/ui/manager/src/components/sidebar/Tree.tsx (hunk 2 above)
Comment (cdedreuille) [reply code/ui/manager/src/components/sidebar/Tree.tsx]: Made the change

File: code/ui/manager/src/components/upgrade/UpgradeBlock.tsx (hunk 1 above)
Comment (cdedreuille) [reply code/ui/manager/src/components/upgrade/UpgradeBlock.tsx]: Made the c hange

**Review by JReinhold — APPROVED** [review]
LGTM!

**Pull request merged**